prowler/providers/aws/services/iam/iam_service.py

```diff
@@ -150,70 +153,70 @@ def __get_password_policy__(self):
+                expiration=password_policy["RequireNumbers"],
+                max_age=max_age,
+                reuse_prevention=reuse_prevention,
+                hard_expiry=hard_expiry,
+            )
         except Exception as error:
             if "NoSuchEntity" in str(error):
                 # Password policy does not exist
-                password_policy = None
+                stored_password_policy = None
             else:
                 logger.error(
                     f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                 )
-        else:
-            return PasswordPolicy(
-                password_policy["MinimumPasswordLength"],
-                password_policy["RequireSymbols"],
-                password_policy["RequireNumbers"],
-                password_policy["RequireUppercaseCharacters"],
-                password_policy["RequireLowercaseCharacters"],
-                password_policy["AllowUsersToChangePassword"],
-                password_policy["ExpirePasswords"],
-                max_age,
-                reuse_prevention,
-                hard_expiry,
-            )
+        finally:
+            return stored_password_policy
 
     def __get_users__(self):
         try:
             get_users_paginator = self.client.get_paginator("list_users")
-        except Exception as error:
-            logger.error(
-                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
-            )
-        else:
             users = []
             for page in get_users_paginator.paginate():
                 for user in page["Users"]:
                     if not self.audit_resources or (
                         is_resource_filtered(user["Arn"], self.audit_resources)
                     ):
                         if "PasswordLastUsed" not in user:
-                            users.append(User(user["UserName"], user["Arn"], None))
+                            users.append(User(name=user["UserName"], arn=user["Arn"]))
                         else:
                             users.append(
                                 User(
-                                    user["UserName"],
-                                    user["Arn"],
-                                    user["PasswordLastUsed"],
+                                    name=user["UserName"],
+                                    arn=user["Arn"],
+                                    password_last_used=user["PasswordLastUsed"],
                                 )
                             )
-
+        except Exception as error:
+            logger.error(
+                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+        finally:
             return users
 
     def __list_virtual_mfa_devices__(self):
         try:
             list_virtual_mfa_devices_paginator = self.client.get_paginator(
```

Comment:
If the `"list_virtual_mfa_devices"` raises an exception we never return the `mfa_devices` because is referenced before assigned.

Reply:
Changed